From f3716c5962d2d182899ce09dd9c0e70d02293ede Mon Sep 17 00:00:00 2001
From: Patrick Rudolph <patrick.rudolph@9elements.com>
Date: Wed, 30 Aug 2023 09:07:38 +0200
Subject: [PATCH] configurations: Add Intel P4510 SSD

The P4510 uses the correct address for NVMe-MI, so add it as
separate configuration and update the regex in p4000 to not
match the P4510.

Change-Id: I5ab3a6e7259cb98940b12c692682335a0f0bac50
Signed-off-by: Patrick Rudolph <patrick.rudolph@9elements.com>
---
 configurations/nvme_p4000.json |  2 +-
 configurations/nvme_p4510.json | 54 ++++++++++++++++++++++++++++++++++
 meson.build                    |  1 +
 3 files changed, 56 insertions(+), 1 deletion(-)
 create mode 100644 configurations/nvme_p4510.json

diff --git a/configurations/nvme_p4000.json b/configurations/nvme_p4000.json
index a75e87a..4860fbb 100644
--- a/configurations/nvme_p4000.json
+++ b/configurations/nvme_p4000.json
@@ -42,7 +42,7 @@
     ],
     "Logging": "Off",
     "Name": "NVMe $index",
-    "Probe": "xyz.openbmc_project.FruDevice({'PRODUCT_PRODUCT_NAME': 'P\\d{4}\\w?'})",
+    "Probe": "xyz.openbmc_project.FruDevice({'PRODUCT_PRODUCT_NAME': 'P(?!4510)\\d{4}\\w?'})",
     "Type": "NVMe",
     "xyz.openbmc_project.Inventory.Decorator.Asset": {
         "Manufacturer": "$PRODUCT_MANUFACTURER",
diff --git a/configurations/nvme_p4510.json b/configurations/nvme_p4510.json
new file mode 100644
index 0000000..01622e3
--- /dev/null
+++ b/configurations/nvme_p4510.json
@@ -0,0 +1,54 @@
+{
+    "Bus": "$bus",
+    "Exposes": [
+        {
+            "Address": "$address",
+            "Bus": "$bus",
+            "Name": "NVMe $index FRU",
+            "Type": "EEPROM"
+        },
+        {
+            "Address": "0x6a",
+            "Bus": "$bus",
+            "Name": "NVMe $index Temp",
+            "Thresholds": [
+                {
+                    "Direction": "greater than",
+                    "Name": "upper critical",
+                    "Severity": 1,
+                    "Value": 115
+                },
+                {
+                    "Direction": "greater than",
+                    "Name": "upper non critical",
+                    "Severity": 0,
+                    "Value": 110
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower non critical",
+                    "Severity": 0,
+                    "Value": 5
+                },
+                {
+                    "Direction": "less than",
+                    "Name": "lower critical",
+                    "Severity": 1,
+                    "Value": 0
+                }
+            ],
+            "Type": "NVME1000"
+        }
+    ],
+    "Logging": "Off",
+    "Name": "NVMe $index",
+    "Probe": "xyz.openbmc_project.FruDevice({'PRODUCT_PRODUCT_NAME': 'P4510\\w?'})",
+    "Type": "NVMe",
+    "xyz.openbmc_project.Inventory.Decorator.Asset": {
+        "Manufacturer": "$PRODUCT_MANUFACTURER",
+        "Model": "$PRODUCT_PRODUCT_NAME",
+        "PartNumber": "$PRODUCT_PART_NUMBER",
+        "SerialNumber": "$PRODUCT_SERIAL_NUMBER"
+    },
+    "xyz.openbmc_project.Inventory.Item.Chassis": {}
+}
diff --git a/meson.build b/meson.build
index 5b9ca0d..c8611b5 100644
--- a/meson.build
+++ b/meson.build
@@ -128,6 +128,7 @@ configs = [
     'mtmitchell_mb.json',
     'nisqually.json',
     'nvme_p4000.json',
+    'nvme_p4510.json',
     'pcie_ssd_retimer.json',
     'pennybacker.json',
     'pssf132202a.json',
--
2.41.0

